### ISSUE TITLE
Args.find doesn't remove item from Args.remains or Args.rest

### ISSUE THREAD

**Issue by zacps** [serenity-rs/serenity#802]
> Search for any available argument that can be parsed, and remove it from the "arguments queue".

> The removal is irreversible. And happens after the search and the parse were successful.

> "Arguments queue" is the list which contains all arguments that were deemed unique as defined by quotations and delimiters. The 'removed' argument can be, likewise, still accessed via message.

This led me to expect that one of `remains` or `rest` would not show the found element in the queue, however they both read from `Args.message` instead of `Args.args` so they still contain the element.

**Comment by arqunis** [comment]
They could. However, in order to accomplish that, would mean paying with an allocation in return for a string of convenience. I didn't want to go through that route, so I chose for them to return a subslice of `message`, which is much cheaper in comparison to the allocation requisite to receive a string without the item that had been removed prior.

But I think we can do better with `Args`' design and prevent betraying expectations of users, at the cost of breaking stuff.
